### PR TITLE
buildFHSEnvBubblewrap: extraPreBwrapCmds after variable initialisation

### PR DESCRIPTION
**Pull request by benaryorg**
## Description of changes

Prior to this commit it was not possible to modify e.g. [the list of ignored directories](https://github.com/NixOS/nixpkgs/blob/7205e5d9c173a2549e95f65e0619a24b8c7adc14/pkgs/build-support/build-fhsenv-bubblewrap/default.nix#L129) at all, however given that `buildFHSEnvBubblewrap` effectively uses a sandboxing tool (*bwrap*) I feel like this is a missed opportunity. The code in nixpkgs already covers all the knobs that are required to get *Nix* itself to run inside bubblewrap, so why not allow users to make that additional modification? While additional `ro_mounts` and such can be *added* to the bubblewrap invocation, the already mounted directories cannot be removed, and even if shadowed by e.g. a tmpfs mount, this would still allow something inside the sandbox to potentially unmount the tmpfs and access the data.

So what this change does is moving the snippet where custom code can be injected down by four lines so that users can actually modify those variables e.g. using `ignored+=( /home /srv /mnt /boot )`. The only cases in which this would break is:

- someone using those variable names in `extraPreBwrapCmds` already and relying on them being overwritten; I would consider that chance slim, and the fix would be easy enough
- someone using a construct like `false && \` to disable the `ignored` initialisation and effectively working around this limitation; again the chances are slim (even though I know [I'd be affected](https://git.shell.bsocat.net/infra/commit/?h=staging&id=568a3a55f7290484a2b946979fb53014ce2f8851)), and the fix would be easy enough (as this change makes the workaround needless anyway so it's an improvement)

TL;DR: this change allows people to use *buildFHSEnvBubblewrap* to build a sandboxing tool like this:

```nix
pkgs.buildFHSEnvBubblewrap
{
  name = "sandbox";
  unshareUser = true;
  unshareIpc = true;
  unsharePid = true;
  unshareUts = true;
  unshareCgroup = true;
  dieWithParent = true;
  extraPreBwrapCmds = "ignored+=( /home /srv /mnt /boot )";
  runScript = "${pkgs.bash}/bin/bash";
  extraBwrapArgs = [ "--dir" "$HOME" "--chdir" "$HOME" "--setenv" "PATH" "${pkgs.nix}/bin" ];
}
```

This will drop you into a shell that has none of the listed directories mounted, and thus is suitable for running untrusted code or anything that you wouldn't want to leave traces in your regular user (programs that drop data into ~/.local/share or other hard to find places, or just anything that you really don't want to have even theoretical access to your SSH private keys but still be able to use `nix shell` or something).
It can be further tuned using the *extraBwrapArgs* of course, but that's out of scope for this PR.

I didn't add a release note because I don't think this is major enough, but I can do so if anyone thinks it would be good.
Also I'm not too invested in this change (considering I have that hacky workaround for now) so feel free to close this if deemed unsuitable.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by philiptaron — APPROVED** [review]
Yeah, I buy it.

**Comment by benaryorg** [comment]
Is there anyone I can bug to get this merged in time for 24.11? (seeing as it's already approved and all)
